src/effects/native/reverbeffect.cpp

```diff
@@ -74,48 +99,35 @@ void ReverbEffect::processChannel(const ChannelHandle& handle,
     Q_UNUSED(handle);
     Q_UNUSED(enableState);
     Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE bandwidth = m_pBandWidthParameter->value();
-    CSAMPLE damping = m_pDampingParameter->value();
 
-    // Flip value around.  Assumes max allowable is 1.0.
-    damping = 1.0 - damping;
+    const auto bandwidth = m_pBandWidthParameter->value();
+    const auto decay = m_pDecayParameter->value();
+    const auto damping = m_pDampingParameter->value();
+    const auto blend = m_pBlendParameter->value();
 
-    bool params_changed = (damping != pState->prev_damping ||
-                           bandwidth != pState->prev_bandwidth);
+    // Set the sample rate (TODO XXX: Maybe we want to call .init() after this?)
+    pState->reverb.setSampleRate(sampleRate);
 
-    pState->reverb.setBandwidth(bandwidth);
-    pState->reverb.setDecay(damping);
+    // Process the buffer using the new current parameter values
+    processBuffer(pState->reverb, pInput, pOutput, numSamples, bandwidth, decay, damping, blend);
+}
 
-    for (uint i = 0; i + 1 < numSamples; i += 2) {
-        CSAMPLE mono_sample = (pInput[i] + pInput[i + 1]) / 2;
+void ReverbEffect::processBuffer(MixxxPlateX2& effect, const CSAMPLE* in, CSAMPLE* out, const unsigned int numSamples,
+                                 const CSAMPLE bandwidth, const CSAMPLE decayParam, const CSAMPLE damping,
+                                 const CSAMPLE blendParam) {
+    effect.setBandwidth(bandwidth);
+    effect.setDamping(damping);
+    const auto decay = .749*decayParam;
```

Comment:
What is .749 here? Can we use a const variable to name it? 
IMHO we should not use auto in this case, because it is not easy to guess which type is decay.
A Bad example in our style guide